### PR TITLE
Offsets store err fix

### PR DESCRIPTION
**Pull request by edenhill**
For https://github.com/confluentinc/confluent-kafka-dotnet/pull/339

### CODE REVIEW

File: src-cpp/rdkafkacpp.h [hunk 1]
```diff
@@ -1718,7 +1718,10 @@ class RD_EXPORT KafkaConsumer : public virtual Handle {
    *
    * @remark \c enable.auto.offset.store must be set to \c false when using this API.
    *
-   * @returns RdKafka::ERR_NO_ERROR on success or an error code on error.
+   * @returns RdKafka::ERR_NO_ERROR on success, or
+   *          RdKafka::ERR___UNKNONW_PARTITION if none of the offsets could
```
Comment (mhowlett) [inline src-cpp/rdkafkacpp.h]: UNKNONW_PARTITION -> UNKNOWN_PARTITION

File: src/rdkafka.h [hunk 2]
```diff
@@ -2367,8 +2367,10 @@ rd_kafka_resp_err_t rd_kafka_offset_store(rd_kafka_topic_t *rkt,
  *
  * @remark \c `enable.auto.offset.store` must be set to "false" when using this API.
  *
- * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success or an error code if
- *          none of the offsets could be stored.
+ * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success, or
+ *          RD_KAFKA_RESP_ERR__UNKNONW_PARTITION if none of the
```
Comment (mhowlett) [inline src/rdkafka.h]: UNKNONW_PARTITION -> UNKNOWN_PARTITION

File: src-cpp/rdkafkacpp.h (hunk 1 above)
Comment (edenhill) [reply src-cpp/rdkafkacpp.h]: good spottage!

File: src/rdkafka_offset.c [hunk 3]
```diff
@@ -715,7 +718,7 @@ rd_kafka_offsets_store (rd_kafka_t *rk,
                 ok_cnt++;
         }
 
-        return offsets->cnt > 0 && ok_cnt < offsets->cnt ?
+        return offsets->cnt > 0 && ok_cnt == 0 ?
```
Comment (mhowlett) [inline src/rdkafka_offset.c]: The alternative is to remove this check altogether. I think I agree what you have is the best option.

**Comment by mhowlett** [comment]
asside from spelling errors, LGTM.

File: src/rdkafka_offset.c (hunk 3 above)
Comment (edenhill) [reply src/rdkafka_offset.c]: This is sort of an optimization on the caller/user's behalf: if all partitions failed the user won't have to check the error code for each individual partition.
Questionable value, but it is there now.

**Pull request merged**